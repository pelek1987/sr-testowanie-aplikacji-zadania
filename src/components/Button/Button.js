import {useState} from "react";

function Button({ label = 'OK' }) {

    const [counter, setCounter] = useState(0);

    const handleClick = () => {
        setCounter(counter + 1);
    }

    return (
        <button onClick={handleClick}>
            { label } <span>{counter}</span>
        </button>
    );
}

export default Button;
