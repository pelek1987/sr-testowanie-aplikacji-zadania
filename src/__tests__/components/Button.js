import TestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import Button from "../../components/Button";

describe('<Button /> testing using snapshots', () => {
    it('renders Button with label "Click me 0"', () => {
        const tree = TestRenderer
            .create(<Button label={'Click me'} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('<Button /> testing using enzyme' , () => {

    let wrapper;

    beforeAll(() => {
        wrapper = shallow(<Button />);
    })
    it('should render ok', () => {
        const button = wrapper.find('button');
        expect(button.text()).toBe('OK 0');
    });

    it('should render "Click me 0" when label="Click me" prop passed', () => {
        wrapper.setProps({label: 'Click me'})
        const button = wrapper.find('button');
        expect(button.text()).toBe('Click me 0');
    });

    it('should have initial counter set to 0', () => {
        const counter = wrapper.find('button span');
        expect(counter.text()).toBe('0');
    })

    it('should increment counter', () => {
        const button = wrapper.find('button');
        let counter = wrapper.find('button span');
        expect(counter.text()).toBe('0');
        button.simulate('click')
        counter = counter = wrapper.find('button span');
        expect(counter.text()).toBe('1');
    });
})
