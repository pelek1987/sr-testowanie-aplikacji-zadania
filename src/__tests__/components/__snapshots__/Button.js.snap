// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> testing using snapshots renders Button with label "Click me 0" 1`] = `
<button
  onClick={[Function]}
>
  Click me
   
  <span>
    0
  </span>
</button>
`;
